.pre-commit-config.yaml

```diff
@@ -1,6 +1,6 @@
 repos:
 - repo: https://github.com/astral-sh/ruff-pre-commit
-  rev: v0.9.9
+  rev: v0.12.0
```

Comment:
I understand pre-commit (the tool) can be useful to some devs to install *git* pre-commit hooks. But considering we don't use it on the CI, we should really consider just making it run a "tox autofix" command to avoid having to duplicate versions here.

Then if we want CI autofixes, https://autofix.ci/ and https://pre-commit.ci/lite both do more or less the same job (pushes any local changes during a CI run)

At that point, `.pre-commit-config.yaml` would only be provided out of convenience for those who enjoy pre-commit hooks and don,t want to manually setup a hook to run said "tox autofix" command.

Ref: https://github.com/jaraco/skeleton/issues/109#issuecomment-2676534591